compiler/qsc_circuit/src/lib.rs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#![warn(clippy::mod_module_files, clippy::pedantic, clippy::unwrap_used)]
+#![allow(clippy::missing_errors_doc, clippy::missing_panics_doc)]
```

Comment:
The pattern for the clippy lints changed since you started working on this, where now instead of adding these to the top of the lib.rs you can just add a workspace lint entry to the Cargo.toml. See https://github.com/microsoft/qsharp/blob/main/compiler/qsc_eval/Cargo.toml#L30-L31 as an example.